### DISCUSSION TITLE
How I can get current url/page ?

### DISCUSSION THREAD

**Discussion by angelside** [hydephp/develop#467] [General]
Hi,

I am using only blade. How can access to current url?

Like below one but for global usage.

`$item->isCurrent()`

Something like this ?

```php
<a href="/" {!! $page->isCurrent() ? 'aria-current="page"' : '' !!}>Homepage</a>
```

**Comment by emmadesilva** [comment]
Hey there! So excited to see that you are trying out HydePHP!

This feature hasn't been documented properly get, so I'll do my best to help. Let me know if you have any questions at all, I'm happy to help!

### How pages/routes work (the long answer)

All pages have a `$page` property, which I see you already discovered. This property always extends the [`AbstractPage`](https://github.com/hydephp/develop/blob/master/packages/framework/src/Contracts/AbstractPage.php) class. Through this class, we have the `getRoute()` method. While HydePHP doesn't use the traditonal routing found in Laravel, we have a kind of pseudo route system where each page has a corresponding route. You can use this to compare a route. 

In addition, all pages also have a `$currentRoute` property which of course contains the current route. We can use this to compare any route/page with the current route. HydePHP uses this in the default navigation bar which I see you also discovered. There, the `$item` is an instance of the `NavItem` class which corresponds to a route. Here's how the `$item->isCurrent()` method is implemented:

```php
public function isCurrent(?PageContract $current = null): bool
{
    if ($current === null) {
        $current = Hyde::currentRoute()->getSourceModel();
    }

    if (! isset($this->route)) {
        return ($current->getRoute()->getRouteKey() === $this->href)
        || ($current->getRoute()->getRouteKey().'.html' === $this->href);
    }

    return $current->getRoute()->getRouteKey() === $this->route->getRouteKey();
}
```

### The short answer

As you can see, we are using the route key, you can use this in your case. So instead of `$page->isCurrent()`, you'd use `$page->getRouteKey() === $currentRoute->getRouteKey()`. Or you should simply be able to use `$page->getRoute() === $currentRoute`!


I hope this answers your question, if not, please let me know as I'm more than happy to help!

**Comment by angelside** [comment]
Hi, thanks for the detailed answer. I am just trying/playing with SSG's. I was actually surprised when I saw all issues belongs to you. I am thinking this gonna be part of a SAAS eventually.


I tried your suggestions:

`{{ $page->getRoute() }}` -> index.html
`{{ $currentRoute->getRouteKey() }}` -> index
`{{ $currentRoute }}` -> index.html

`$page->getRoute() === $currentRoute` -> FALSE (that is weird why is false)
`$page->getRoute() == $currentRoute` -> TRUE

```
@if ($page->getRoute() == $currentRoute)
	ALWAYS TRUE
@endif
```

I've finally end with this, if I keep navigation in a array and loop it, it's gonna be more clean.

```
<a href="/" class="{!! ($currentRoute->getRouteKey() == 'index') ? 'text-red-500' : '' !!}">
    Homepage
</a>
```

**Reply by emmadesilva** [reply]
Yeah, I use GitHub issues to keep track if a lot of stuff like todos :)

Weird, that === doesn't work, but glad == works. It might be because `$page->getRoute()` and `$currentRoute` both return a Route object, but any time you use it as a string (like in `{{ }}`) it gets converted into a relative link to that page. Try running `@dd($page->getRoute(), $currentRoute)` to see the full types.

Let me know if you have any more questions!
